syncd/pfc_detect_mlnx.lua

```diff
@@ -0,0 +1,102 @@
+-- KEYS - queue IDs
+-- ARGV[1] - counters db index
+-- ARGV[2] - counters table name
+-- ARGV[3] - poll time interval
+-- return queue Ids that satisfy criteria
+
+local counters_db = ARGV[1]
+local counters_table_name = ARGV[2]
+local poll_time = tonumber(ARGV[3])
+
+local rets = {}
+
+redis.call('SELECT', counters_db)
+
+-- Iterate through each queue
+local n = table.getn(KEYS)
+for i = n, 1, -1 do
+    local counter_keys = redis.call('HKEYS', counters_table_name .. ':' .. KEYS[i])
+    local pfc_duration_key = ''
+    local pfc_rx_pkt_key = ''
+    local counter_num = 0
+    local old_counter_num = 0
+    local is_deadlock = false
+    local pfc_wd_status = redis.call('HGET', counters_table_name .. ':' .. KEYS[i], 'PFC_WD_STATUS')
+    if pfc_wd_status == 'operational' then
+        local detection_time = tonumber(redis.call('HGET', counters_table_name .. ':' .. KEYS[i], 'PFC_WD_DETECTION_TIME'))
+        local time_left = redis.call('HGET', counters_table_name .. ':' .. KEYS[i], 'PFC_WD_DETECTION_TIME_LEFT')
+        if not time_left  then
+            time_left = detection_time
+        else
+            time_left = tonumber(time_left)
+        end
+
+        local m = table.getn(counter_keys)
+        for j = m, 1, -1 do
+            if string.sub(counter_keys[j],-string.len('RX_PKTS'))=='RX_PKTS' then
```

Comment:
>string.sub(counter_k [](start = 15, length = 20)

change to the exact match?

Reply:
I don't know the exact name of stat ID, becasuse it is something like SAI_PORT_STAT_PFC_[i]_RX_PKTS